src/QuadVelocityController.cpp

```diff
@@ -109,13 +116,52 @@ bool QuadVelocityController::getVelocities(geometry_msgs::Twist& return_velociti
     ros::Time time = ros::Time::now();
 
     // Check if we already have a velocity at this time
-    if (time == last_velocity_stamped_.header.stamp) {
+    if (have_last_velocity_stamped_ && time == last_velocity_stamped_.header.stamp) {
         return_velocities = last_velocity_stamped_.twist;
         return true;
     }
 
+    // We don't have a velocity at this time yet, so make sure we don't fetch
+    // a transform at the same time as the last one
+    while (ran_once_ && time == last_transform_stamped_.header.stamp) {
```

Comment:
The purpose was to update the pid controllers with info from the current timestamp. However it can be changed to what you said.

Should this blocking loop call ros::OK and spinonce? Could simulated time mess with it if you don't call those functions?